scripts/upload_ci_stats.sh

```diff
@@ -149,6 +158,20 @@ do
     mv "$x.tmp" "$x"
 done
 
+# Save a snippet of logs where a backend was terminated by signal.
+grep -C40 "was terminated by signal" postgres.log > postgres-failure.log ||:
+
+# Find internal program errors in Postgres logs.
+awk -v OFS='\t' '
+    match($0, /XX000:\s+(.+)$/, a) { error = a[1]; next; }
+    !error { next; }
+    match($0, /LOCATION:\s+(.+)$/, a) { location = a[1]; next; }
+    match($0, /STATEMENT: (.+)$/, a) { statement = a[1]; next; }
+    (statement && match($0, /^\s+(.*)$/, a)) { statement = statement " " a[1]; next; }
+    statement { print ENVIRON["JOB_DATA"], error, location, statement; error = ""; statement = ""; location = ""}
+' < postgres.log > ipe.tsv ||:
```

Comment:
We have JSON logs since PG15, so maybe we can enable them and convert this into a somewhat less ugly `jq` invocation.